### PR TITLE
BWS-24: + bs and space creation page

### PR DESCRIPTION
**Pull request by Donymak**
image
image

### CODE REVIEW

File: frontend/src/components/app/app.tsx [hunk 1]
```diff
@@ -16,6 +16,8 @@ import { EAM } from 'components/eam/eam';
 import { EAMWorkerCreate } from 'components/eam-create-worker/eam-create-worker';
 import { EAMConfigurateGroup } from 'components/eam-configurate-group/eam-configurate-group';
 import { NotFound } from 'components/not-found-page/not-found-page';
+import { BS } from '../bs/bs';
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]:
```suggestion
import { BS } from 'components/bs/bs';
```

File: frontend/src/components/app/app.tsx [hunk 2]
```diff
@@ -16,6 +16,8 @@ import { EAM } from 'components/eam/eam';
...
+import { BSCreateSpace } from '../bs-create-space/bs-create-space';
```
Comment (what1s1ove) [inline frontend/src/components/app/app.tsx]:
```suggestion
import { BSCreateSpace } from 'components/bs-create-space/bs-create-space';
```

File: frontend/src/components/bs-create-space/bs-create-space.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
+import { FC } from 'react';
+import styles from './bs-create-space.module.scss';
+import { Button } from '../common/button/button';
```
Comment (what1s1ove) [inline frontend/src/components/bs-create-space/bs-create-space.tsx]:
```suggestion
import { Button } from 'components/common/common';
```
fix & merge all imports

File: frontend/src/components/bs/bs.tsx [hunk 4]
```diff
@@ -0,0 +1,22 @@
+import { FC } from 'react';
+import styles from './styles.module.scss';
+import { Button } from '../common/button/button';
+import { AppRoute } from '../../common/enums/app/app-route.enum';
```
Comment (what1s1ove) [inline frontend/src/components/bs/bs.tsx]: fix imports

File: frontend/src/store/bs-space/actions.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  BSSpaceCreateRequestDto,
+  AsyncThunkConfig
+>(
+  ActionType.CREATE_SPACE,
+  async (payload: BSSpaceCreateRequestDto, { extra }) => {
+    const { bsApi, navigation, notification } = extra;
+
+    const space = await bsApi.createSpace({
+      ...payload,
+    });
```
Comment (what1s1ove) [inline frontend/src/store/bs-space/actions.ts]:
```suggestion
    const space = await bsApi.createSpace(payload);
```

File: frontend/src/store/bs-space/actions.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
```
Comment (what1s1ove) [inline frontend/src/store/bs-space/actions.ts]: this is part of the creation page. so name it as `frontend/src/store/bs-space-create/actions.ts`

File: frontend/src/store/bs-space/common.ts [hunk 7]
```diff
@@ -0,0 +1,5 @@
+enum ActionType {
+  CREATE_SPACE = 'bs/create-space',
```
Comment (what1s1ove) [inline frontend/src/store/bs-space/common.ts]:
```suggestion
  CREATE_SPACE = 'bs-create-space/create',
```

File: frontend/src/store/root-reducer.ts [hunk 8]
```diff
@@ -4,11 +4,13 @@ import { reducer as eam } from './eam/reducer';
 import { reducer as toastr } from 'react-redux-toastr';
 import { reducer as EAMGroupConfigurate } from './eam-group-configurate/reducer';
 import { reducer as EAMWorkerConfigurate } from './eam-worker-configurate/reducer';
+import { reducer as bs } from './bs-space/reducer';
```
Comment (what1s1ove) [inline frontend/src/store/root-reducer.ts]:
```suggestion
import { reducer as BSSpaceCreate } from './bs-space-create/reducer';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/store/bs-space-create/reducer.ts [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+import { createSpace } from './actions';
+
+type State = {
+  dataStatus: DataStatus;
+  spaces: BSSpaceCreateResponseDto[];
+};
+
+const initialState: State = {
+  dataStatus: DataStatus.IDLE,
+  spaces: [],
```
Comment (what1s1ove) [inline frontend/src/store/bs-space-create/reducer.ts]:
```suggestion
```
you don't need all spaces on this page

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by Yaroslaww-1 — DISMISSED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**